src/main/java/duke/logic/Parser.java

```diff
@@ -0,0 +1,275 @@
+                }
+                if (response.endsWith(" /at")) {
+                    throw new IllegalTimeException("No time specified.");
+                }
+                int tokenPosition = response.indexOf(" /at ");
+                if (tokenPosition < 0) {
+                    throw new IllegalTokenException("No token found.");
+                }
+                String description = response.substring(0, tokenPosition);
+                String time = response.substring(tokenPosition + 5);
+                //to ensure whitespaces alone as a description or time is not allowed
+                if (description.strip().length() == 0) {
+                    throw new IllegalDescriptionException("No description specified.");
+                }
+                if (time.length() == 0) {   //should not even come here as if time is pure whitespace, will be truncated
+                    throw new IllegalTimeException("No time specified.");
+                }
+                //strip leading get rid of white spaces infront, back alr dealt with
+                time = time.stripLeading();
+                return new EventCommand(taskList, description, Parser.parseTime(time));
+            }
+        } catch (IllegalDescriptionException e) {
+            return () -> {
+                System.out.println(" ☹ OOPS!!! The description cannot be empty.");
+            };
+        } catch (IllegalTimeException e) {
+            return () -> {
+                System.out.println(" ☹ OOPS!!! The time specified is either invalid or empty.");
+            };
+        } catch (IllegalTokenException e) {
+            return () -> {
+                System.out.println(" ☹ OOPS!!! You are mising the \"/by\" or \"/at\" token.");
+            };
+        }
+        return () -> {
+            System.out.println(" ☹ OOPS!!! I'm sorry, but I don't know what that means. :-(");
+        };
+    }
+    
+    /**
+     * Parses string to the appropriate time it represents, if any.
+     *
+     * @param str the string to parse.
+     * @return the LocalDate object representing time.
+     * @throws IllegalTimeException If unable to parse the string to a valid time.
+     */
+    private static LocalDate parseTime(String str) throws IllegalTimeException {
+        String sDay;
+        String sMonth;
+        String sYear;
+        //ddmmyyyy
+        String[] arr = new String[3];
+        //forward slash, hyphen, dot, or whitespace, otherwise no separator cases
+        if (str.indexOf('/') >= 0) {
+            arr = Parser.parseTimeHelper('/', str);
+        } else if (str.indexOf('-') >= 0) {
+            arr = Parser.parseTimeHelper('-', str);
+        } else if (str.indexOf('.') >= 0) {
+            arr = Parser.parseTimeHelper('.', str);
+        } else if (str.indexOf(' ') >= 0) {
+            arr = Parser.parseTimeHelper(' ', str);
+        } else if (str.length() == 4 || str.length() == 6 || str.length() == 8) {
```

Comment:
Would `parseTimeDelimited` be a better naming since `Helper` sounds more general so it conveys less detail on what it actually does?